### ISSUE TITLE
[MDEP-491] maven-dependency-plugin ignores -Dmaven.local.repo setting

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#933]
**[Eric B](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=benze)** opened **[MDEP-491](https://issues.apache.org/jira/browse/MDEP-491?redirect=false)** and commented

If a maven build is launched with a custom local repo defined, the local repo is not inherited/understood by the dependency plugin.

Ex:
mvn -Dmaven.local.repo=c:\temp\repo clean package -X

I see the following in the debug output
[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-dependency-plugin:2.10:copy' with basic configurator -->
[DEBUG]   (s) groupId = ca.tbt
[DEBUG]   (s) artifactId = tbt-applet
[DEBUG]   (s) version = 2.0.1-SNAPSHOT
[DEBUG]   (s) groupId = ca.tbt
[DEBUG]   (s) artifactId = tbt-val-applet
[DEBUG]   (s) version = 2.0.1-SNAPSHOT
[DEBUG]   (s) artifactItems = [ca.tbt:tbt-applet:2.0.1-SNAPSHOT:jar, ca.tbt:tbt-val-applet:2.0.1-SNAPSHOT:jar]
[DEBUG]   (s) local =       id: local
url: file:///C:/Users/C45841A6/.m2/repository/
layout: default
snapshots: [enabled => true, update => always]
releases: [enabled => true, update => always]

As you can see, the local repo is pointing to the default ~/.m2/repository repo.

Consequently, any artifacts generated in my custom local repo are not available to the plugin and the build fails.

Defining -DlocalRepositoryDirectory has no impact

---

**Affects:** 2.10

**Comment by jira-importer** [comment]
**[Eric B](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=benze)** commented

I noticed in your debug trace that you are using the dependency plugin 2.8.  Have you confirmed this is fixed in > 2.10?  I filed this bug a couple of years ago and do not remember all the specifics but do see that it was against 2.10 at the time.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Indeed, see

```
[DEBUG] Configuring mojo org.apache.maven.plugins:maven-dependency-plugin:3.0.0:tree from plugin realm ClassRealm[plugin>org.apache.maven.plugins:maven-dependency-plugin:3.0.0, parent: sun.misc.Launcher$AppClassLoader@55f96302]
[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-dependency-plugin:3.0.0:tree' with basic configurator -->
[DEBUG]   (f) appendOutput = false
[DEBUG]   (f) outputType = text
[DEBUG]   (f) project = MavenProject: org.apache.maven.shared:maven-filtering:3.1.2-SNAPSHOT @ D:\Entwicklung\Projekte\maven-filtering\pom.xml
[DEBUG]   (f) session = org.apache.maven.execution.MavenSession@ecfbe91
[DEBUG]   (s) skip = false
[DEBUG]   (f) tokens = standard
[DEBUG]   (f) verbose = false
[DEBUG] -- end configuration --
```

but with 2.10:

```
[DEBUG] Configuring mojo org.apache.maven.plugins:maven-dependency-plugin:2.10:tree from plugin realm ClassRealm[plugin>org.apache.maven.plugins:maven-dependency-plugin:2.10, parent: sun.misc.Launcher$AppClassLoader@55f96302]
[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-dependency-plugin:2.10:tree' with basic configurator -->
[DEBUG]   (f) appendOutput = false
[DEBUG]   (f) localRepository =       id: local
      url: file:///D:/Entwicklung/Projekte/maven-filtering/repo/
   layout: default
snapshots: [enabled => true, update => always]
 releases: [enabled => true, update => always]

[DEBUG]   (f) outputType = text
[DEBUG]   (f) project = MavenProject: org.apache.maven.shared:maven-filtering:3.1.2-SNAPSHOT @ D:\Entwicklung\Projekte\maven-filtering\pom.xml
[DEBUG]   (s) skip = false
[DEBUG]   (f) tokens = standard
[DEBUG]   (f) verbose = false
[DEBUG] -- end configuration --
```

But this parameter has been removed in [r1687641](http://svn.apache.org/r1687641) due to [MDEP-494](http://svn.apache.org/r1686634). This all looks correct to me now.
